### PR TITLE
[SPARK-10763][ML][JAVA][TEST] Update Java MLLIB/ML tests to use simplified dataframe construction

### PR DESCRIPTION
**Pull request by holdenk**
As introduced in https://issues.apache.org/jira/browse/SPARK-10630 we now have an easier way to create dataframes from local Java lists. Lets update the tests to use those.

### CODE REVIEW

**Comment by srowen** [comment]
It seems OK to me. I assume we still have at least one test that tests the version with an RDD argument!

**Comment by holdenk** [comment]
@srowen So this changes it so that all of the ml tests that created a Dataframe using an RDDs of Rows now use List's of rows as input to the Dataframe constructor, but in the Dataframe constructor tests both are tested.

File: mllib/src/test/java/org/apache/spark/ml/feature/JavaBucketizerSuite.java [hunk 1]
```diff
@@ -55,16 +55,16 @@ public void tearDown() {
...
-      RowFactory.create(-0.3),
-      RowFactory.create(0.0),
-      RowFactory.create(0.2)
-    ));
     StructType schema = new StructType(new StructField[] {
       new StructField("feature", DataTypes.DoubleType, false, Metadata.empty())
     });
-    DataFrame dataset = jsql.createDataFrame(data, schema);
+    DataFrame dataset = jsql.createDataFrame(
+        Arrays.asList(
```
Comment (mengxr) [inline mllib/src/test/java/org/apache/spark/ml/feature/JavaBucketizerSuite.java]: fix indentation. (It seems that your format configuration for Java is different from Scala.)

**Comment by mengxr** [comment]
LGTM except indentation. The test for creating DataFrame from JavaRDD should live in DataFrame's unit tests.

**Comment by mengxr** [comment]
Merged into master. Thanks!

**Pull request closed without merging**